### DISCUSSION TITLE
Shorten working directory in smaller windows

### DISCUSSION THREAD

**Discussion by jeevanshah07** [JanDeDobbeleer/oh-my-posh#5434] [Troubleshoot] [answered]
I recently switched from p10k to oh my posh and I've found that I can replicate almost my entire previous config with omp, however, I was wondering if there's a way to shorten the current directory path when the window is of smaller width. 

For example (in p10k)  a terminal of full width would have a prompt like this:
![full](https://github.com/user-attachments/assets/c66c2d70-29cb-4ae5-9866-9fb22cf08625)

and a terminal of smaller width displays like this:
![half](https://github.com/user-attachments/assets/8952902a-a84c-4720-baa8-862fdd6122f5)

however, in omp I haven't found a way to replicate this and my prompt gets all jumbled in smaller widths:
![image](https://github.com/user-attachments/assets/a5a17205-843b-4127-8f0a-22870c749e20)

does anyone know how to help?

**Comment by v-amorim** [comment, marked as answer]
You can create two path segments, set the [style][path_style] differently on each and use [segment][segment]'s `min_width` and `max_width` to trigger which one you want based on the terminal size.

[path_style]: https://ohmyposh.dev/docs/segments/system/path#style
[segment]: https://ohmyposh.dev/docs/configuration/segment

**Reply by JanDeDobbeleer** [reply]
@v-amorim holy moly. Even I didn't think about this one. That's genius.

**Reply by v-amorim** [reply]
@JanDeDobbeleer that's nice to hear haha, I did this to show/hide segments whenever the terminal was too big/small, but it left my .json too cluttered, so I dropped it and remade my entire theme to be compact

**Reply by Moneda-Radek** [reply]
any chance you could share the code?
I tried that and ended up with two very similar looking segments
